contracts/ts/genMaciState.ts

```diff
@@ -405,35 +366,37 @@ const genMaciStateFromContract = async (
         numSrQueueOps,
       },
     });
-  }
+  });
 
-  for (const log of mergeMessageAqLogs) {
-    assert(log != undefined);
-    const mutableLogs = {
-      ...log,
-      topics: [...log.topics],
-    };
+  mergeMessageAqLogs.forEach((log) => {
+    assert(!!log);
+    const mutableLogs = { ...log, topics: [...log.topics] };
     const event = pollIface.parseLog(mutableLogs);
 
-    const messageRoot = BigInt(event.args._messageRoot);
+    const messageRoot = BigInt((event?.args as unknown as { _messageRoot: string })._messageRoot);
     actions.push({
       type: "MergeMessageAq",
       blockNumber: log.blockNumber,
       transactionIndex: log.transactionIndex,
       data: { messageRoot },
     });
-  }
+  });
 
   // Sort actions
-  sortActions(actions);
+  actions = sortActions(actions);
 
   // Reconstruct MaciState in order
 
-  for (const action of actions) {
-    if (action["type"] === "SignUp") {
-      maciState.signUp(action.data.pubKey, action.data.voiceCreditBalance, action.data.timestamp);
-    } else if (action["type"] === "DeployPoll") {
-      if (action.data.pollId === pollId) {
+  actions.forEach((action) => {
+    switch (true) {
+      case action.type === "SignUp": {
+        const { pubKey, voiceCreditBalance, timestamp } = action.data;
+
+        maciState.signUp(pubKey!, BigInt(voiceCreditBalance!), BigInt(timestamp!));
+        break;
+      }
+
+      case action.type === "DeployPoll" && action.data.pollId === pollId: {
```

Comment:
Here there seems to be an issue when running the cli via `ts-node`: 

debug:
```ts
console.log("action.data.poll.id === pollId",action.data.pollId === pollId)
console.log("typeof action.data.pollId", typeof action.data.pollId)
console.log("typeof pollId", typeof pollId)
console.log("pollId", pollId)
console.log("action.data.pollId", action.data.pollId) 
```

Output:

```ts
action.data.poll.id === pollId false
typeof action.data.pollId number
typeof pollId string
pollId 0
action.data.pollId 0
```

This happens even though the type of `pollId` is a number

Please refer to https://gist.github.com/ctrlc03/153834129cf478aa50936bfa77271cb6 - the genProofs command will fail here: https://github.com/privacy-scaling-explorations/maci/blob/60dd75fcbf3a8db1e0f57f37bb3f0690f21467da/contracts/ts/genMaciState.ts#L449